### PR TITLE
Add column type information to table creation call

### PR DESCRIPTION
**Pull request by waxlamp**
This PR makes use of https://github.com/multinet-app/multinetjs/pull/22.

This uses the updated multinetjs library to send column type information at table upload time.

Depends on https://github.com/multinet-app/multinetjs/pull/22 being merged and released.
Depends on #136.

### CODE REVIEW

File: src/utils/files.ts [hunk 1]
```diff
@@ -46,10 +46,16 @@ interface TypeScore {
...
 }
 
-async function csvFileTypeRecommendations(file: File): Promise<Map<string, CSVColumnType>> {
-  const parsePromise: Promise<Map<string, CSVColumnType>> = new Promise((resolve) => {
+interface Recommendation {
+  typeRecs: Map<string, CSVColumnType>;
+  rowSample: Array<{}>;
+}
+
+async function csvFileTypeRecommendations(file: File): Promise<Recommendation> {
```
Comment (AlmightyYakob) [inline src/utils/files.ts:54]: If we're going to be returning multiple things from here, I feel like we should rename these things. The interface could be something like `CSVAnalysisResult`, and the function something like `AnalyzeCSV`. I'm not sure if those are the best names, but we're doing two different things here now, and I think it'd be good to reflect that in the function/type names.

File: src/components/TableDialog.vue [hunk 2]
```diff
@@ -139,6 +185,22 @@ export default defineComponent({
     },
   },
   setup(props, { emit }) {
+    // Stepper control
+    const step: Ref<number> = ref(1);
+    const rowSample: Ref<Array<{}>> = ref([]);
```
Comment (AlmightyYakob) [inline src/components/TableDialog.vue:190]: It's a minor thing, but I think a better name for this is `sampleRows`. I feel like that better conveys that this is just meant to be a sample of rows from the whole file. A comment may be good too.

Feel free to push back if you think it's too minor of a change.

File: src/components/TableDialog.vue [hunk 3]
```diff
@@ -139,6 +185,22 @@ export default defineComponent({
...
+        return [];
+      }
+
+      const keys = Object.keys(rowSample.value[0]);
+
+      return keys.map((key) => ({
+        text: key,
+        value: key,
+      }));
+    });
```
Comment (AlmightyYakob) [inline src/components/TableDialog.vue:202]: I think this looks just a bit cleaner:
```suggestion
    const headers = computed(() => {
      const keys = Object.keys(rowSample.value[0] || {});
      return keys.map((key) => ({
        text: key,
        value: key,
      }));
    });
```

**Comment by AlmightyYakob** [comment]
I think this looks really great! Just left a few suggestions

**Comment by waxlamp** [comment]
> Another suggestion I just thought of, which we'll likely want to do as a follow up:
> 
> Since we've now got this type selection UI, we could add a `primaryKey` type to our system, which would replace the following section in the first step:
> 
> Screenshot from 2020-11-19 14-39-59
> 
> We could remove this option from the first step, and just have it as another type in the column drop down in the second step. We would also likely default that to the `_key` field if it's there, or just the first column if it's not there.
> 
> There's important distinction from the rest of the types however, which is that there can only be one primary key column. We'd have to add some logic when it comes to selecting that type, and we'd need to create a new validation error on the server to handle multiple supplied primary keys, but those tasks seem fairly low-effort.
> 
> Since this involves both server and client changes, we should probably address it in a separate issue, and move forward with this PR. However, I thought it was worthy of discussion.
> 
> @waxlamp lmk what you think.

I think this is a good idea and I agree that we should address it in a followup. Can you an issue and link it back here?

File: src/components/TableDialog.vue (hunk 2 above)
Comment (waxlamp) [reply src/components/TableDialog.vue]: See 137f1671e635636d08d00e3b0da614191b96cc1d.

File: src/utils/files.ts (hunk 1 above)
Comment (waxlamp) [reply src/utils/files.ts]: Good idea; check out the names I chose: d7ef1fbd31555f78e04484abcb544b653aa5ea4f.

**Review by AlmightyYakob — APPROVED** [review]

**Review by AlmightyYakob — COMMENTED** [review]
While I approve of these changes, I'm holding off my official approval until the proper version of `multinetjs` is listed.

**Review by AlmightyYakob — APPROVED** [review]
:shipit:

**Pull request merged**